great_expectations/dataset/pandas_dataset.py

```diff
@@ -527,6 +527,9 @@ def expect_column_values_to_not_be_in_set(self, column, value_set,
                                               mostly=None,
                                               parse_strings_as_datetimes=None,
                                               result_format=None, include_config=False, catch_exceptions=None, meta=None):
+        if value_set is None:
```

Comment:
We had to address the "vacuously true" issue while developing 0.7.0 and 0.7.1. This was driven primarily by adding the profiling feature. Profilers work by adding expectations that must be vacuously true. They apply no constraints. These expectations are used only to compute the observed value the profiler can display. We have been using `None` as the value of the "constraining arguments" (e.g., `value`, `value_set`, `min_value`, `max_value`, etc.) to mean "anything is acceptable".

Your implementation is correct if `value_set=None` is interpreted as "expect column values to __not__ belong to the universal set". Then the expectation is vacuously false. 

However, to be consistent with the convention we started following for all other expectation types, let's treat `None` as a way to say that anything is acceptable and change the line to `return np.ones(len(column), dtype=np.bool_)`

Reply:
We should definitely figure out a place to include this explanation in the docs.

It's a helpful/necessary convention for profiling, but it wouldn't be at all obvious if you don't understand the reasoning behind it.